llvm/tools/llvm-readobj/ELFDumper.cpp

```diff
@@ -2086,21 +2086,34 @@ template <typename ELFT> void ELFDumper<ELFT>::parseDynamicTable() {
       break;
     case ELF::DT_RELR:
     case ELF::DT_ANDROID_RELR:
+    case ELF::DT_AARCH64_AUTH_RELR:
       DynRelrRegion.Addr = toMappedAddr(Dyn.getTag(), Dyn.getPtr());
       break;
     case ELF::DT_RELRSZ:
     case ELF::DT_ANDROID_RELRSZ:
+    case ELF::DT_AARCH64_AUTH_RELRSZ:
       DynRelrRegion.Size = Dyn.getVal();
-      DynRelrRegion.SizePrintName = Dyn.d_tag == ELF::DT_RELRSZ
-                                        ? "DT_RELRSZ value"
-                                        : "DT_ANDROID_RELRSZ value";
+      if (Dyn.d_tag == ELF::DT_RELRSZ)
```

Comment:
Probably this should be another `switch`.

Reply:
For this case, I think it is better to do

```
if (emachine == EM_AARCH64 && xxx == ELF::DT_AARCH64_AUTH_RELRSZ) {
  DynRelrRegion.Size = Dyn.getVal();
  DynRelrRegion.SizePrintName = "DT_AARCH64_AUTH_RELRSZ value";
  continue;
}
switch (Dyn.d_tag) {
  ...
}
```

If emachine is not EM_AARCH64, `DT_AARCH64_AUTH_RELRSZ` is not defined.

Reply:
Agreed with @MaskRay.

Reply:
Moved checks for `DT_AARCH64_AUTH_RELRSZ` and `DT_AARCH64_AUTH_RELRENT` to an `if` before the main `switch` as you suggested. These two values are distinguished inside the `if` via inner `switch` with `continue` at the end of each `case`.